src/nuget.rs

```diff
@@ -0,0 +1,286 @@
+
+  <Target Name="_CheckForUnsupportedPlatformTarget"
+          Condition="'$(EnableUnsupportedPlatformTargetCheck)' == 'true'"
+          AfterTargets="_CheckForInvalidConfigurationAndPlatform">
+
+    <!--
+    Special case .NET Core portable applications.  When building a portable .NET Core app,
+    the PlatformTarget is empty, and you don't know until runtime (i.e. which dotnet.exe)
+    what processor architecture will be used.
+    -->
+    <Error Condition="('$(PlatformTarget)' != 'x64' AND '$(PlatformTarget)' != 'x86') AND
+                      ('$(OutputType)' == 'Exe' OR '$(OutputType)'=='WinExe') AND
+                      !('$(TargetFrameworkIdentifier)' == '.NETCoreApp' AND '$(PlatformTarget)' == '')"
+           Text="{} currently supports 'x64' and 'x86' processor architectures. Please ensure your application is targeting 'x64' or 'x86'." />
+  </Target>
+  
+</Project>"#,
+            package_id
+        )
+    }
+
+    pub fn add_nuspec(&mut self, nuspec: &Nuspec) -> io::Result<()> {
+        let nuspec_file_name = format!("{}.nuspec", nuspec.metadata.id);
+        let nuspec_content = nuspec.to_xml();
+        self.zip
+            .start_file(&nuspec_file_name, FileOptions::default())?;
+        self.zip.write_all(nuspec_content.as_bytes())?;
+        Ok(())
+    }
+
+    pub fn add_files(&mut self, project: &Project) -> io::Result<()> {
+        for platform_dir in &project.platform_directories {
+            if !(matches!(platform_dir.os, Os::Linux | Os::Macos | Os::Windows)
+                && matches!(platform_dir.cpu, Cpu::X86_64 | Cpu::Aarch64))
+            {
+                continue;
+            }
+            for loadable_file in &platform_dir.loadable_files {
+                let os = match platform_dir.os {
+                    Os::Windows => "win",
+                    Os::Linux => "linux",
+                    Os::Macos => "osx",
+                    Os::Android => "android",
+                    Os::Ios => "ios",
+                    Os::IosSimulator => "ios-simulator",
+                };
+                let cpu = match platform_dir.cpu {
+                    Cpu::Aarch64 => "arm64",
+                    Cpu::X86_64 => "x64",
+                    Cpu::I686 => "x86",
+                    Cpu::Armv7a => "armv7",
+                };
+                let file_path =
+                    format!("runtimes/{}-{}/native/{}", os, cpu, loadable_file.file.name);
+                self.zip.start_file(&file_path, FileOptions::default())?;
+                self.zip.write_all(&loadable_file.file.data)?;
+            }
+        }
+        Ok(())
+    }
+
+    pub fn add_cs_file(&mut self, project: &Project) -> io::Result<()> {
+        let package_name = project.spec.targets.nuget.as_ref().map_or_else(
+            || project.spec.package.name.replace('-', "_"),
+            |nuget| nuget.friendly_name.clone(),
+        );
+        let cs_file_content = format!(
+            r#"namespace Microsoft.Data.Sqlite
```

Comment:
This should remove various code styling/analyzer warnings/suggestions from the included source file.

```suggestion
            r#"// <auto-generated/>

namespace Microsoft.Data.Sqlite
```